src/components/book-content-mixin.cjsx

```diff
@@ -29,6 +33,78 @@ module.exports =
       else
         img.onload = sizeImage
 
+  getCNXIdOfHref: (href) ->
+    _.first(url.split('#'))
+
+  getMediaTag: (media) ->
+    # form media tag text based on tag name or data-type
+    tag = media.tagName
+    tag = media.dataset.type if media.dataset.type?
+
+    # capitalize
+    # TODO find one place for all such string formating helpers to go
+    tag = tag.charAt(0).toUpperCase() + tag.substring(1).toLowerCase()
+
+  buildReferenceBookLink: (cnxId) ->
+    referenceBookParams = _.clone(@context.router.getCurrentParams())
+    referenceBookParams.cnxId = cnxId or @getCNXId()
+    pageUrl = @context.router.makeHref('viewReferenceBookPage', referenceBookParams)
+
+    pageUrl
+
+  isMediaLink: (link) ->
+    link.innerText is '[link]' and link.hash.length > 0 and link.hash.search('/') is -1
+
+  hasCNXId: (link) ->
+    trueHref = link.getAttribute('href')
+    link.hash.length > 0 and trueHref.substr(0, 1) isnt '#'
+
+  shouldOpenNewTab: ->
+    # Make link external only if this is not already a ref book page.
+    @constructor.displayName isnt 'ReferenceBookPage'
```

Comment:
Why rely on the `displayName` instead of `ReferenceBookPage::shouldOpenNewTab: -> true`?